### PR TITLE
[CSDiagnostics] Offer a fix-it to insert a return type placeholder when returning from a void function

### PR DESCRIPTION
**Pull request by theblixguy**
Currently, when we return from a `Void`-returning function, an error is emitted:

```swift
func voidFunc() {
  return 0 // error: unexpected non-void return value in void function
}
```

It's likely that the user forgot to add a return type, so add a fix-it to insert a placeholder for the type.

### CODE REVIEW

File: lib/Sema/CSDiagnostics.cpp [hunk 1]
```diff
@@ -4792,6 +4792,21 @@ bool InvalidUseOfAddressOf::diagnoseAsError() {
...
+    // have an explicit return type. The reason we also need to check
+    // whether the parameter list has a valid loc is to guard against
+    // cases like like 'var foo: () { return 1 }' as here that loc will
+    // be invalid.
+    if (FD->getBodyResultTypeLoc().getLoc().isInvalid() &&
+        FD->getParameters()->getStartLoc().isValid()) {
+      auto fixItLoc = Lexer::getLocForEndOfToken(
+          getASTContext().SourceMgr, FD->getParameters()->getEndLoc());
+      emitDiagnostic(anchor->getLoc(), diag::add_return_type_note)
+          .fixItInsert(fixItLoc, " -> <# Return Type #>");
```
Comment (xedin) [inline lib/Sema/CSDiagnostics.cpp]: Let's jam it all together here so placeholder is `<#ReturnType#>` :)

**Review by xedin — APPROVED** [review]

**Comment by theblixguy** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@theblixguy Let's hold off from triggering CI for now, it would have to be re-triggered after LLVM changes are done anyway.

**Comment by theblixguy** [comment]
@swift-ci please smoke test

**Pull request merged**